src/model.jl

```diff
@@ -1,106 +1,275 @@
-function BatchNormWrap(out_ch)
-    Chain(x->expand_dims(x,2),
-	  BatchNorm(out_ch),
-	  x->squeeze(x))
-end
+using Flux
+using Flux: @functor
 
-UNetConvBlock(in_chs, out_chs, kernel = (3, 3)) =
-    Chain(Conv(kernel, in_chs=>out_chs,pad = (1, 1);init=_random_normal),
-	BatchNormWrap(out_chs),
-	x->leakyrelu.(x,0.2f0))
+struct ConvBlock
```

Comment:
This should be parameterised

```suggestion
struct ConvBlock{T}
```

Reply:
done